ipn/ipnlocal/local.go

```diff
@@ -1645,9 +1650,8 @@ func (b *LocalBackend) SetControlClientStatus(c controlclient.Client, st control
 			prefs.Persist = st.Persist.AsStruct()
 		}
 	}
-	if st.LoginFinished() {
-		if b.authURL != "" {
-			b.resetAuthURLLocked()
+	if st.LoggedIn {
+		if authWasInProgress {
```

Comment:
Not saying it's necessarily logically incorrect, but moving the reset to the block above and using the cached view of the auth URL isn't a no-op change, as something else could write to the authURL while we were unlocked.

Has this been thoroughly tested for race conditions?